### ISSUE TITLE
Argument type deduction failed when passing address of an i32 struct field to a function accepting *i32 argument.

### ISSUE THREAD

**Issue by asahi3g** [skycoin/cx#214]
**Describe the bug**
Argument type deduction failed when passing address of an i32 struct field to a function accepting *i32 argument (case 2).
No compilation error when passing address of a i32 var (case 1).

**To Reproduce**
```package main

type test struct {
	i i32
}

func foo(pi *i32) (i i32) {
	i = *pi
}

func main()() {
	var i i32 = 44
	test(foo(&i), 44, "") // case 1

	var t test
	t.i = 55
	test(foo(&t.i), 55, "") // case 2
}
```

**Expected behavior**
No compilation error.

**Screenshots**
```
error: pending_3.cx:17 function 'foo' expected input argument of type '*i32'; 'i32' was provided
```

**Desktop (please complete the following information):**
 - OS: Linux
 - CX Version: 0.5.18

**Additional context**

**Comment by ingwal** [comment]
Hmm, why is this an error?  t.i is an i32, so &t.i should be a *i32.  Or is the precedence rules such that '&t.i' should be read as (&t).i rather than &(t.i)?  In the latter case, the parameter would of course be wrong.

**Comment by amherag** [comment]
Fixed in #236.
